packages/core/src/App/Containers/Redirect/redirect.jsx

```diff
@@ -136,7 +136,10 @@ const Redirect = observer(() => {
                 }
             }
 
-            setResetTradingPasswordModalOpen(true);
+            // The if statement is needed because the modal from core is triggered by the stores.
+            // Whereas the one in wallet is triggered by the hooks.
+            // This is to prevent the modal from being triggered twice.
+            if (!routes.wallets) setResetTradingPasswordModalOpen(true);
```

Comment:
Can't we do something without touching code outside of wallets package?

Reply:
I don't think so. The values are set here and stored in localStorage. In the wallets, we simply check if they're there. Adding this if statement won't prompt the store to open the ResetPasswordModal.

Reply:
what about the same pattern which is used in: /deriv-app/packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

Reply:
We still need to use this regardless because core keeps the `verifcation_code` inside.